configure

```diff
@@ -382,17 +382,17 @@ if [ -z "$ANTIC_LIB" ]; then
    case "$OS" in
       Darwin)
          ANTIC_LIBNAME="libantic.dylib"
-	 ANTIC_LIB="libantic-$ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH.dylib"
-         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
+         ANTIC_LIB="libantic.$ANTIC_MAJOR.dylib"
+         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```

Comment:
```suggestion
         EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$ANTIC_LIB -compatibility_version $ANTIC_MAJOR.$ANTIC_MINOR -current_version $ANTIC_MAJOR.$ANTIC_MINOR.$ANTIC_PATCH";;
```

Actually, I don't think that antic is following semantic versioning @isuruf so let's keep it at x.x. That's also what our antic-feedstock over at conda-forge does.

Reply:
Then the SONAME in Linux should reflect that too. Currently the SONAME uses only `MAJOR_VERSION`

Reply:
@isuruf I guess you're right. So my (limited) understanding is that we should compile into
`libantic.so.major.minor.patch` and then symlink `libantic.so` and the `libantic.so.major.minor`.

That's the situation on Linux.

What's the correct thing to do on macOS and Windows? Essentially the same? I.e.,

Windows; Compile into `libantic-major.minor.patch.dll` and "symlink" this to "`libantic.dll` and `libantic-major.minor.dll`? (It seems to me that Windows does not do versioning in .dll names usually; at least I could not find much about it online.)

macOS: Compile into `libantic.major.minor.patch.dylib` and symlink there from `libantic.dylib` and `libantic.major.minor.dylib`?

Reply:
Or is the only important thing that we set the `-soname` flags correctly and we should not worry too much about the naming of the files?

Reply:
> So my (limited) understanding is that we should compile into
libantic.so.major.minor.patch and then symlink libantic.so and libantic.so.major.minor.

Yes. SONAME should be then `libantic.so.major.minor`. Similar for macOS.

On windows, directly compile into `libantic-major.minor.dll` and no symlinks.